lib/rabbit_feed/event.rb

```diff
@@ -2,12 +2,19 @@ module RabbitFeed
   class Event
     include ActiveModel::Validations
 
-    attr_reader :schema, :payload
-    validates_presence_of :schema, :payload
+    attr_reader :schema, :payload, :application, :host, :environment, :created_at_utc, :version, :name
+    validates :schema, :application, :host, :environment, :created_at_utc, :version, :name, presence: true
+    validates :payload, length: { minimum: 0, allow_nil: false, message: 'can\'t be nil' }
 
-    def initialize schema, payload
-      @schema  = schema
-      @payload = payload
+    def initialize schema, payload, metadata
+      @schema         = schema
+      @payload        = payload.with_indifferent_access.freeze if payload
+      @application    = metadata['application']
```

Comment:
the event for me would have a single `headers` property that could point to anoither `Header` object that has these properties on it. then you can get them with `event.headers`